orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -425,6 +575,83 @@ class TaskController {
     )
   }
 
+  @PostAuthorize("hasPermission(returnObject.application, 'APPLICATION', 'READ')")
+  private List<String> getPipelineConfigIdsOfReadableApplications() {
+    List<String> applicationNames = front50Service.getAllApplications()*.name as List<String>
+    List<String> pipelineConfigIds = applicationNames.stream()
+      .map{ applicationName -> front50Service.getPipelines(applicationName, false)*.id as List<String> }
+      .flatMap{ c -> c.stream() }
+      .collect(Collectors.toList())
+
+    return pipelineConfigIds
+  }
+
+  private static boolean recursivelyCheckIfObjectFieldsMatchesAllSubsetFields(Object object, Object subset) {
```

Comment:
This function seems complex enough to warrant its own test case in a spec.

Reply:
I'd make this name a bit terser, e.g. `checkObjectMatchesSubset` and describe the recursion algo a bit in a javadoc comment.